docs/aws/audit/esmonitoring/rules/es_ri_recent_purchases.mdx

```diff
@@ -23,6 +23,252 @@ CBP
+   ```
+
+3. **Set Up Budget Alerts:**
+   Create budget alerts to notify you when spending on Reserved Instances exceeds a certain threshold. This helps in keeping track of unexpected purchases.
+
+   ```sh
+   aws budgets create-budget --account-id <account-id> --budget '{"BudgetName": "ElasticsearchReservedInstanceBudget", "BudgetLimit": {"Amount": "1000", "Unit": "USD"}, "TimeUnit": "MONTHLY", "BudgetType": "COST"}' --notifications-with-subscribers '[{"Notification": {"NotificationType": "ACTUAL", "ComparisonOperator": "GREATER_THAN", "Threshold": 80, "ThresholdType": "PERCENTAGE", "NotificationState": "ALARM"}, "Subscribers": [{"SubscriptionType": "EMAIL", "Address": "your-email@example.com"}]}]'
+   ```
+
+4. **Tagging Reserved Instances:**
+   Implement a tagging strategy for your Reserved Instances to easily identify and manage them. This can help in reviewing and auditing recent purchases.
+
+   ```sh
+   aws es add-tags --arn <reserved-instance-arn> --tag-list Key=Environment,Value=Production Key=Owner,Value=TeamA
+   ```
+
+By following these steps, you can effectively prevent misconfigurations related to Elasticsearch Reserved Instance recent purchases using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Elasticsearch Reserved Instance recent purchases from being misconfigured, you can use Python scripts to automate the review and management of these instances. Below are the steps to achieve this:
+
+### 1. **Set Up AWS SDK for Python (Boto3)**
+First, you need to install and set up Boto3, the AWS SDK for Python, to interact with AWS services.
+
+```bash
+pip install boto3
+```
+
+### 2. **Authenticate and Initialize Boto3 Client**
+Create a Python script to authenticate and initialize the Boto3 client for Elasticsearch.
+
+```python
+import boto3
+
+# Initialize a session using Amazon Elasticsearch
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+es_client = session.client('es')
+```
+
+### 3. **Fetch and Review Reserved Instances**
+Write a function to fetch and review the recent purchases of Elasticsearch Reserved Instances.
+
+```python
+def list_reserved_instances():
+    response = es_client.describe_reserved_elasticsearch_instances()
+    reserved_instances = response.get('ReservedElasticsearchInstances', [])
+    
+    for instance in reserved_instances:
+        print(f"Reserved Instance ID: {instance['ReservedElasticsearchInstanceId']}")
+        print(f"Instance Type: {instance['ElasticsearchInstanceType']}")
+        print(f"Start Time: {instance['StartTime']}")
+        print(f"Duration: {instance['Duration']} seconds")
+        print(f"State: {instance['State']}")
+        print(f"Fixed Price: {instance['FixedPrice']}")
+        print(f"Usage Price: {instance['UsagePrice']}")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>